drivers/sensor/nxp_kinetis_temp/temp_kinetis.c

```diff
@@ -98,11 +98,11 @@ static int temp_kinetis_channel_get(struct device *dev,
 		slope_uv = config->slope_hot_uv;
 	}
 
-	/* m x 1000 */
-	m = (adcr_vdd * slope_uv) / vdd_mv;
+	adcr_1000m = (adcr_vdd * slope_uv) / vdd_mv;
 
 	/* Temperature in milli degrees Celsius */
-	temp_mc = 25000 - ((data->buffer[0] - adcr_temp25) * 1000000) / m;
+	temp_mc = 25000 -
+		(((data->buffer[0] - adcr_temp25)) * 1000 / adcr_1000m) * 1000;
```

Comment:
You can lose precision by moving the divide earlier like this. Unless there's a risk of overflow in the multiply by 1000000, I don't recommend this change.

Reply:
There is a risk of overflow when multiplying with 1000000.

Reply:
I will try to come up with a solution without risk of overflow where we do not loose as much precision.

Reply:
@MaureenHelm Please take a look at my updated solution.